netbout-web/src/main/java/com/netbout/rest/MarkdownTxtmark.java

```diff
@@ -37,9 +40,51 @@
  * @since 2.23
  */
 public final class MarkdownTxtmark implements Markdown {
+    /**
+     * Plain link detection pattern.
+     */
+    private static final Pattern LINK = Pattern.compile(
+        "https?://[a-zA-Z0-9-._~:/\\?#@!$&'*+,;=%]+[a-zA-Z0-9-_~/#@$&'*+=%]"
+    );
 
     @Override
     public String html(@NotNull final String txt) {
-        throw new UnsupportedOperationException("not implemented yet");
+        return Processor.process(MarkdownTxtmark.formatLinks(txt));
+    }
+
+    /**
+     * Replace plain links with Markdown syntax. To be convinced it doesn't
+     * replace links inside markdown syntax, it ensures that characters
+     * before and after link do not match to Markdown link syntax.
+     * @param txt Text to find links in
+     * @return Text with Markdown-formatted links
+     */
+    private static String formatLinks(final String txt) {
```

Comment:
@dmzaytsev this method looks very complex ... I believe, this could be done with pure replaceAll with proper regexp pattern.
Have you tried this? If this solution was rejected, please, tell why (just curiosity :) )